packages/expo-modules-core/ios/ModuleRegistry.swift

```diff
@@ -14,6 +14,10 @@ public final class ModuleRegistry: Sequence {
    */
   internal func register(holder: ModuleHolder) {
     log.info("Registering module '\(holder.name)'")
+    // module already registered, don't re-register
+    if registry[holder.name] != nil {
+      return
+    }
```

Comment:
@tsapeta - this is the piece of this PR that needs the most attention. see note in summary.

Reply:
Honestly this seems like it should have been there all along -- or alternatively, throw if the system tries to register the same module twice so we can understand why.

Reply:
The problem is a lot of expo go relies upon overriding modules (I think, unconfirmed if whether it's by this mechanism or another).

Reply:
I think it's fine for now, but could you please elaborate more in the comment above (specifically mention that it's necessary for expo-updates and Expo Go) and maybe do `log.warn` that says that the module is already registered and was not overridden.

In the future, we can think what we can do to make it better. Maybe the `register(module:)` could take another param like `canBeOverridden` that we set in the ModuleHolder and then skip re-registering when it's set to `false` (`true` by default)

Reply:
I think I'll make that `canBeOverridden` change now to reduce the possible scope of this PR.